From 0f80828d73856be223cdc9878f7c096ffe58759f Mon Sep 17 00:00:00 2001
From: David Sommerseth <davids@openvpn.net>
Date: Fri, 23 May 2025 10:11:19 +0200
Subject: [PATCH] tests: Upgrade to googletest-1.17.0-1

This fixes an issue when compiling with GCC-15 and newer.  This adds a
related compiler warning fix to the machine-id unit test as well.

Signed-off-by: David Sommerseth <davids@openvpn.net>
---
 src/tests/unit/machine-id.cpp |  1 +
 subprojects/gtest.wrap        | 22 ++++++++++++++--------
 2 files changed, 15 insertions(+), 8 deletions(-)

diff --git a/src/tests/unit/machine-id.cpp b/src/tests/unit/machine-id.cpp
index 8131947..a9df81c 100644
--- a/src/tests/unit/machine-id.cpp
+++ b/src/tests/unit/machine-id.cpp
@@ -14,6 +14,7 @@
 
 #include "build-config.h"
 
+#include <iomanip>
 #include <iostream>
 #include <fstream>
 #include <string>
diff --git a/subprojects/gtest.wrap b/subprojects/gtest.wrap
index ca5d699..e6f1686 100644
--- a/subprojects/gtest.wrap
+++ b/subprojects/gtest.wrap
@@ -1,10 +1,16 @@
 [wrap-file]
-directory = googletest-release-1.10.0
+directory = googletest-1.17.0
+source_url = https://github.com/google/googletest/archive/refs/tags/v1.17.0.tar.gz
+source_filename = gtest-1.17.0.tar.gz
+source_hash = 65fab701d9829d38cb77c14acdc431d2108bfdbf8979e40eb8ae567edf10b27c
+patch_filename = gtest_1.17.0-1_patch.zip
+patch_url = https://wrapdb.mesonbuild.com/v2/gtest_1.17.0-1/get_patch
+patch_hash = 79f438b1844fa5f2ef002b56065a5ee2f7d473add4230fbc0a888ca197e6f1ed
+source_fallback_url = https://github.com/mesonbuild/wrapdb/releases/download/gtest_1.17.0-1/gtest-1.17.0.tar.gz
+wrapdb_version = 1.17.0-1
 
-source_url = https://github.com/google/googletest/archive/release-1.10.0.zip
-source_filename = gtest-1.10.0.zip
-source_hash = 94c634d499558a76fa649edb13721dce6e98fb1e7018dfaeba3cd7a083945e91
-
-patch_url = https://wrapdb.mesonbuild.com/v1/projects/gtest/1.10.0/1/get_zip
-patch_filename = gtest-1.10.0-1-wrap.zip
-patch_hash = 04ff14e8880e4e465f6260221e9dfd56fea6bc7cce4c4aff0dc528e4a2c8f514
+[provide]
+gtest = gtest_dep
+gtest_main = gtest_main_dep
+gmock = gmock_dep
+gmock_main = gmock_main_dep
